shared/fs/header/header-container.js

```diff
@@ -2,41 +2,57 @@
 import * as Types from '../../constants/types/fs'
 import {compose, connect, setDisplayName, type Dispatch} from '../../util/container'
 import {fsTab} from '../../constants/tabs'
-import {navigateAppend, navigateTo} from '../../actions/route-tree'
+import {navigateAppend, navigateTo, navigateUp} from '../../actions/route-tree'
+import {isMobile} from '../../constants/platform'
 import FolderHeader from './header'
 
 type OwnProps = {
   path: Types.Path,
 }
 
+type DispatchProps = {
+  onOpenBreadcrumb: (path: string) => (evt?: SyntheticEvent<>) => void,
+  _onOpenBreadcrumbDropdown: (
+    dropdownItems: Array<Types.PathBreadcrumbItem>,
+    isTeamPath: boolean,
+    onOpenBreadcrumb: (path: string) => (evt?: SyntheticEvent<>) => void
+  ) => (evt?: SyntheticEvent<>) => void,
+}
+
 const mapStateToProps = () => ({})
 
 const mapDispatchToProps = (dispatch: Dispatch) => ({
-  onOpenBreadcrumb: (path: string) =>
-    dispatch(navigateTo([fsTab, {props: {path: Types.stringToPath(path)}, selected: 'folder'}])),
-  onOpenBreadcrumbDropdown: (
+  onOpenBreadcrumb: (path: string) => (evt?: SyntheticEvent<>) => {
```

Comment:
lets not do this. just use mergeProps to export a function which takes the evt and passes that down into dispatchProps_onOpenBreadcrumb

Reply:
Sure, but isn't that effectively the same thing?

Reply:
no. one is a function we're exporting and another is a function that you call that gives you another function. we don't use this pattern anywhere